databricks_blueprints/download_file_from_dbfs.py

```diff
@@ -0,0 +1,134 @@
+import argparse
+import os
+import sys
+import base64
+import requests
+import re
+import shipyard_utils as shipyard
+try:
+    import helpers
+    import errors
+except BaseException:
+    from . import helpers
+    from . import errors
+
+ONE_MB = 1024*1024
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--access-token', dest='access_token', required=True)
+    parser.add_argument('--instance-id', dest='instance_id', required=True)
+    parser.add_argument('--source-file-name', dest='source_file_name', required=True)
+    parser.add_argument('--source-folder-name', dest='source_folder_name', required=False)
+    parser.add_argument('--dest-file-name', dest='dest_file_name', required=False)
+    parser.add_argument('--dest-folder-name', dest='dest_folder_name', required=False)
+    parser.add_argument('--source-file-name-match-type', dest='source_file_name_match_type',
+                        choices={'exact_match', 'regex_match'}, required=True)
+    args = parser.parse_args()
+    return args
+
+
+def download_file_from_dbfs(client, source_file_path, dest_file_path):
+    read_handle_endpoint = "/dbfs/read"
+    # check if filetype works with binary or not
+    if ('.csv' in source_file_path) or ('.txt' in source_file_path):
+        is_binary = False
+        write_mode = "w+"
+    else:
+        is_binary = True
+        write_mode = 'wb+'
+    with open(dest_file_path, write_mode) as file:
+        offset = 0
+        while True:
+            payload = {
+                "path": source_file_path,
+                "offset": offset, 
+                "length": ONE_MB # 1MB of data 
+            }
+            try:
+                read_response = client.get(read_handle_endpoint, params=payload)
+            except Exception:
+                print(f'Connection Error for {read_handle_endpoint}')
+                print(f"Data downloaded: {offset/1024}MB")
```

Comment:
Why are you dividing by 1024 and (1024*1024) or just printing the offset directly?

Reply:
Do you suggest i just leave the downloaded data in terms of bytes or should i remove that information completely from the output?

Reply:
I think leaving it in bytes is fine. [](edited-with-Axolo)